### PR TITLE
minica: use buildGoModule

### PR DESCRIPTION
**Pull request by azahi**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by m1cr0man** [comment]
@ofborg test acme

File: pkgs/tools/security/minica/default.nix [hunk 1]
```diff
@@ -1,33 +1,32 @@
...
     repo = "minica";
     rev = "v${version}";
-    sha256 = "18518wp3dcjhf3mdkg5iwxqr3326n6jwcnqhyibphnb2a58ap7ny";
+    sha256 = "sha256-3p6rUFFiWXhX9BBbxqWxRoyRceexvNnqcFCyNi5HoaA=";
   };
 
-  ldflags = [
-    "-X main.BuildVersion=${version}"
-  ];
+  vendorSha256 = "sha256-pQpattmS9VmO3ZIQUFn66az8GSmB4IvYhTTCFn6SUmo=";
```
Comment (m1cr0man) [inline pkgs/tools/security/minica/default.nix:17]: When should or shouldn't you add vendorSha256? I noticed in #177203 it was set to null.

**Review by m1cr0man — APPROVED** [review]
Looks good. Accidently started the acme test twice 😛 but if that passes I'm happy to see it merged.

File: pkgs/tools/security/minica/default.nix (hunk 1 above)
Comment (azahi) [reply pkgs/tools/security/minica/default.nix:17]: When the source repository already has a `vendor` directory (i.e. dependencies are already bundled), then it should be set to `null`. If not, we pull dependencies as an input and calculate their SHA.

**Comment by azahi** [comment]
Regarding ACME tests, they fail for me on the `master` branch without my patch. So, I think this must be an unrelated issue.

**Comment by m1cr0man** [comment]
Oh sorry, yes. Someone already informed me of that issue on Matrix. We can go ahead and merge, I'll fix anything that's still broken later.

File: pkgs/tools/security/minica/default.nix [hunk 2]
```diff
@@ -1,33 +1,32 @@
...
   src = fetchFromGitHub {
     owner = "jsha";
     repo = "minica";
     rev = "v${version}";
-    sha256 = "18518wp3dcjhf3mdkg5iwxqr3326n6jwcnqhyibphnb2a58ap7ny";
+    sha256 = "sha256-3p6rUFFiWXhX9BBbxqWxRoyRceexvNnqcFCyNi5HoaA=";
   };
 
-  ldflags = [
-    "-X main.BuildVersion=${version}"
```
Comment (SuperSandro2000) [inline pkgs/tools/security/minica/default.nix:17]: Missing -s -w

**Pull request merged**